packages/esm-patient-orders-app/src/lab-results/lab-results-form-field.component.tsx

```diff
@@ -41,153 +40,109 @@ const ResultFormField: React.FC<ResultFormFieldProps> = ({ concept, control, def
               />
-            )}
-            {isCoded(member) && (
-              <Controller
-                name={member.uuid}
+            ) : isCodedField ? (
+              <Select
+                {...field}
+                className={styles.textInput}
+                defaultValue={defaultValue?.value?.uuid}
+                id={`select-${concept.uuid}`}
+                key={concept.uuid}
+                labelText={labelText}
+                invalidText={error?.message}
+                invalid={Boolean(error?.message)}
+              >
+                <SelectItem text={t('chooseAnOption', 'Choose an option')} value="" />
+                {concept?.answers?.length &&
+                  concept?.answers?.map((answer) => (
+                    <SelectItem key={answer.uuid} text={answer.display} value={answer.uuid}>
+                      {answer.display}
+                    </SelectItem>
+                  ))}
+              </Select>
+            ) : null
+          }
+        />
+        {isPanelField
+          ? concept.setMembers.map((member) => (
+              <ResultFormField
+                key={member.uuid}
+                concept={member}
                 control={control}
-                render={({ field }) => (
-                  <Select
-                    {...field}
-                    className={styles.textInput}
-                    defaultValue={getSavedMemberValue(member.uuid, member.datatype.display)}
-                    id={`select-${member.uuid}`}
-                    key={member.uuid}
-                    labelText={`${member?.display ? member.display + ' ' : ''}${formatLabRange(member)}`}
-                    type="text"
-                    invalidText={errors[member.uuid]?.message}
-                    invalid={!!errors[member.uuid]}
-                  >
-                    <SelectItem text={t('chooseAnOption', 'Choose an option')} value="" />
-
-                    {member?.answers?.map((answer) => (
-                      <SelectItem key={answer.uuid} text={answer.display} value={answer.uuid}>
-                        {answer.display}
-                      </SelectItem>
-                    ))}
-                  </Select>
-                )}
+                defaultValue={getSavedMemberDefaultObservation(member.uuid)}
               />
-            )}
-          </React.Fragment>
-        ))}
+            ))
+          : null}
+      </>
+    );
+  }
+
+  return (
+    <>
+      <label className={styles.label}>{labelText}</label>
+      <InlineNotification
+        kind="error"
+        hideCloseButton
+        title={t('invalidConceptDatatype', 'Invalid concept datatype')}
+        subtitle={t(
+          'updateConceptDatatypeFromNa',
+          'Update concept datatype from {{datatype}} to appropriate datatype',
+          {
+            datatype: concept.datatype.display,
+            interpolation: { escapeValue: false },
+          },
+        )}
```

Comment:
I'm not too sure that this message explains why the current datatype is invalid or that the suggested guidance is clear enough to end users. 

The best alternative I could come up with for now is potentially too verbose:

"Invalid lab test configuration: This test needs to be configured with a specific type (like number, text, or choice list) to record results properly. Please contact your system administrator to fix this."